### PR TITLE
Add classes and methods 

### PR DESCRIPTION
**Pull request by Bettenoch**
<h2 dir="auto">Made the following changes:♻</h2>

🔹Create class Person with the following:

- Instance vars: @id, @name, and @age.
- Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
- Getters for @id, @name, and @age.
- Setters for @name and @age.
- Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.
- Public method can_use_services? that returns true if person is of age or if they have permission from parents.

🔸Create class Student with the following:

- Inherits from Person.
- Constructor extends parent's constructor by adding @classroom and a parameter for it.
- Method play_hooky that returns "¯\(ツ)/¯".

🔹Create class Teacher with the following:

- Inherits from Person.
- Constructor extends parent's constructor by adding @specialization and a parameter for it.
- Override can_use_services? so it always returns true.

<hr></hr>
🕹
<h3 dir="auto">THANK YOU FOR REVEIWING MY PR:🙏</h3>

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require './person'
+
+class Student < Person
+  attr_accessor :classroom
+
+  # constructor method
+  def initialize(classroom)
```
Comment (zeff96) [inline student.rb]: - [ ] Hello! 🌟 In order to ensure that the Student class's initialize method accepts age, name (with a default value of "unknown"), and parent_permission (with a default value of true), it's important to pass these parameters to both the Student class's initialize method and the super call within the Student class. By doing so, you're ensuring that the Student class inherits all the attributes from its parent class, Person. 🚀📚

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+require './person'
+
+class Teacher < Person
+  # getter/setter method
+  attr_accessor :specialization
+
+  # constructor method
+  def initialize(specialization)
```
Comment (zeff96) [inline teacher.rb]: Hello! 🌟 In order to ensure that the Teacher class's initialize method accepts age, name (with a default value of "unknown"), and parent_permission (with a default value of true), it's important to pass these parameters to both the Teacher class's initialize method and the super call within the Teacher class. By doing so, you're ensuring that the Teacher class inherits all the attributes from its parent class, Person. 🚀📚

File: README.md [hunk 3]
```diff
@@ -1,2 +1,198 @@
...
+ 
+```
+
+### Install
+
+Install this project with:
+
+```sh
+  cd OOP-school=library
+  node npm install
```
Comment (zeff96) [inline README.md]: - [ ] Hello! 🐦 Do Ruby projects require an npm install to install packages? Let's take a closer look and provide a more detailed explanation.

- Ruby projects typically use the RubyGems package manager to manage and install Ruby-related libraries and dependencies. You use commands like gem install to add Ruby gems to your project.

- On the other hand, npm is the package manager for JavaScript and Node.js projects, not Ruby. If you have JavaScript dependencies or front-end assets (like CSS frameworks or JavaScript libraries) in your Ruby project, you might need to run npm install to manage those specific dependencies, but this is separate from managing Ruby gems.

- So, in summary, while Ruby projects primarily use RubyGems for managing Ruby-related dependencies, npm install is typically used for JavaScript and Node.js-related dependencies within a Ruby project, such as those used for the front end. 📦💎🔍

File: README.md [hunk 4]
```diff
@@ -1,2 +1,198 @@
...
+To run the project, execute the following command:
+run live server
+
+### Run tests
+
+To run tests, run the following command:
+Run Github Actions Test
+
+```sh
+  npm test
```
Comment (zeff96) [inline README.md]: - [ ] Hey there! 🚀 Do you need to use npm test to run tests for Ruby projects? Let's dive in and clarify.

- In general, Ruby projects use testing frameworks like RSpec or MiniTest, which are specifically designed for Ruby code. To run these tests, you typically use commands like rspec or rake test, depending on the testing framework you're using.

- On the other hand, npm test is a command used in JavaScript and Node.js projects to run tests written in JavaScript, often using frameworks like Jest or Mocha. It's not directly related to Ruby testing.

- So, in summary, for Ruby tests, you would use Ruby-specific testing commands, not npm test. 🧪💎🔍

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️ ♻️ ♻️ 
Hi @Bettenoch , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

close

## Highlights
✔️ Linters passing
✔️ Correct GitFlow
✔️ Descriptive pull request

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @Bettenoch 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

image


Highlights
- [x] Good project organization ✔️
- [x] Nice use of Ruby features ✔️  
- [x] Kudus on meeting the requirements  as suggested by the last reviewer ✔️


Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).
